services/src/layers/error.rs

```diff
@@ -0,0 +1,25 @@
+use super::listing::LayerCollectionId;
+use crate::api::model::datatypes::LayerId;
+use snafu::Snafu;
+
+#[derive(Debug, Snafu)]
+#[snafu(
+    visibility(pub(crate)),
+    context(suffix(false)) /* disables default `Snafu` suffix */,
+)]
+pub enum LayerDbError {
+    #[snafu(display("There is no layer with the given id {id}"))]
+    NoLayerForGivenId { id: LayerId },
+
+    #[snafu(display("There is no layer collection with the given id {id}"))]
+    NoLayerCollectionForGivenId { id: LayerCollectionId },
+
+    #[snafu(display("There is no layer {layer} in collection {collection}"))]
+    NoLayerForGivenIdInCollection {
```

Comment:
that's kind of weird that this error can happen. If layers can be in multiple collections, then they shuld be accessible by their layerId only right? Smells like an error in the API/concept.